packages/@sanity/cli/templates/shopify/components/hotspots/ProductTooltip.tsx

```diff
@@ -0,0 +1,26 @@
+import {useSchema} from 'sanity'
+import {Box} from '@sanity/ui'
+import {HotspotTooltipProps} from 'sanity-plugin-hotspot-array'
+
+interface HotspotFields {
+  productWithVariant?: {
+    product: {
+      _ref: string
+    }
+  }
+}
+
+export default function ProductPreview({value, renderPreview}: HotspotTooltipProps<HotspotFields>) {
```

Comment:
Nitpick (feel free to ignore): I think this could be more readable like below. Also, we try to avoid using inline styling and prefer using a styled components outside the component scope

```tsx
import styled from 'styled-components'
import {useSchema} from 'sanity'
import {Box} from '@sanity/ui'
import {HotspotTooltipProps} from 'sanity-plugin-hotspot-array'

const StyledBox = styled(Box)`
  width: 200px;
`

export default function ProductPreview(props: HotspotTooltipProps<HotspotFields>) {
  const {value, renderPreview} = props
  const productSchemaType = useSchema().get('product')
  const hasProduct = value?.productWithVariant?.product?._ref && productSchemaType

  const previewProps = useMemo(
    () => ({
      value: value?.productWithVariant?.product,
      schemaType: productSchemaType,
      layout: 'default',
    }),
    [productSchemaType, value?.productWithVariant?.product]
  )

  return (
    <StyledBox padding={2}>
      {hasProduct ? renderPreview(previewProps) : `No product selected`}
    </StyledBox>
  )
}

```

Reply:
Leaving this for now as I have other pressing items to attend to and this is triggering some TS errors. May revisit.